docs/source/developers/java/building.rst

```diff
@@ -32,8 +32,12 @@ Arrow Java uses the `Maven <https://maven.apache.org/>`_ build system.
 
 Building requires:
 
+* JDK 22+
```

Comment:
is it 21 or 22? (preferably we would at least build with the latest LTS?)

Reply:
Would like to test with 22, one of the latest Java features I'd like see Arrow adopt is [JEP 454 - Foreign Function and Memory API](https://openjdk.org/jeps/454) which was released as part of Java 22

Reply:
Testing with 22 is fine, but I'd like the minimum version to at least be an LTS

Reply:
Is it possible to have an optionally-enabled submodule for FFM experiments?

Reply:
Yes, it is possible but what about the release process? wouldn't that require Java 22?

Reply:
I think it's not a problem to build with Java 22 for the release, I'm just thinking there may be people who would build JARs themselves instead of relying on Maven and would not necessarily want to use a non-LTS release

Reply:
Maybe this is not a the strongest idea in this context, but for test purpose, can we keep 21 as the latest, and use preview features to do the testing, and later we can upgrade as we get LTS?

Reply:
I agree with @lidavidm and @vibhatha and with an eye on reproducibility (and potential future patch releases) we should build on LTS for the release as well.

Reply:
For test purpose it is still possible to use Java 21 (or 17 or 11) for testing. This was basically the purpose of #43380.

I guess I was pushing a bit my luck (or trying to provoke feedback, and I thank you all for it) by going with the latest release JDK vs LTS. I think we can adjust when I (or someone else) would be proposing changes which would leverage the recent APIs introduced in Java 22. That said, are we all comfortable in requiring latest LTS (currently 21) to build Arrow or would prefer the status-quo?

Reply:
21 Latest would be fine IMHO.

Reply:
> I agree with @lidavidm and @vibhatha and with an eye on reproducibility (and potential future patch releases) we should build on LTS for the release as well.

This would basically mean delaying any (optional) use of FFM/FFI to Java 25 so not before september 2025. I'm also not sure I follow on the reproducibility requirement. @assignUser can you clarify what do you mean by that?

Reply:
@laurentgo I understand this concern, and I am also keen on using FFM in the memory layer. My plan is (at present) to keep it as an optional or preview like version for the users (I could be wrong and no discussions have started yet, maybe we can start here). And the argument for keeping an LTS version is what I personally believe considering the broader community adoption of the library.

Reply:
Let me just reemphasize that this only changes the JDK version required to build Arrow, not the JDK version to use Arrow (which is still Java 11). Leveraging FFM would be done through the [MRJAR](https://openjdk.org/jeps/238) mechanism which allows to embed multiple versions of the same class tailored for specific Java version.

Reply:
>  delaying any (optional) use of FFM/FFI to Java 25 so not before september 2025.

hm, I see. The main issue I see is when we need to re-build the binaries either for a patch release or other reasons, if we originally release with a non lts version and this version e.g. 22 is no longer available additional changes will be required that potentially have other issues that need to be fixed (e.g. dependencies). vs. just using the lts that will continue to be available for much longer.

Though this is extrapolated from previous patch/backport experience and not actually based on java, so maybe my fears are unfounded?

Reply:
As far as docker images or adoptium/temurin availability of non-LTS version. it seems older versions are still available. It is to be the same for the Github `setup-java` action. The issue may be homebrew though

Reply:
I'm OK packaging with 22, I just want to leave it open for people to build with 21 (or ideally, even 17) if needed, minus any FFM code.

Reply:
Ok. So for now I think I will pause on this change and will wait until I'm ready to propose a change requiring the use of a newer Java version

Reply:
@laurentgo what would be the expected changes with this PR, if we break it down to the list of proposed changes? 
(Removing UNSAFE, etc)